src/streamlink/plugins/twitch.py

```diff
@@ -595,8 +607,8 @@ def _check_for_rerun(self):
             return False
 
         try:
-            stream_type, broadcast_platform, broadcaster_software = self.api.stream_rerun(self.channel_id)
-            if stream_type != "live" or broadcast_platform == "rerun" or broadcaster_software == "watch_party_rerun":
+            stream_type = self.api.stream_rerun(self.channel)
+            if stream_type == "rerun":
```

Comment:
```suggestion
            if stream_type != "live":
```

This is better, as Twitch has changed the stream type names several times, and using the GQL API doesn't ensure that it'll stay "rerun". "live" however will certainly not change.